ExampleMod/Common/GlobalItems/WeaponWithGrowingDamage.cs

```diff
@@ -0,0 +1,176 @@
+
+			//Example: UpdateValue() is called when a stack is combined.
+			//		The original item.stack was 10, the new item.stack is 100.
+			//		The original experience was 40, the current experience is 40.
+			//		The original extra coins from selling would be 
+			//
+			//	Previous Values:
+			//			
+			//			lastBonusValue: 200
+			//			stack: 10
+			//	New Values:
+			//			
+
+			if (stack == int.MinValue)
+				stack = item.stack + stackChange;
+
+			if (CheckStackZero(item))
+				return;
+
+			int lastBonusValuePerItem = lastBonusValue / stack;//lastBonusValue : 200, stack: 10, lastBonusValuePerItem = 20
+			stack = item.stack + stackChange;//item.stack = 10, stackChange = 90, so the new stack will be 100.  stack = 100
+
+			if (CheckStackZero(item))
+				return;
+
+			int newBonusValue = experience * 5;//experience: 40, newBonusValue = 200
+			int newBonusValuePerItem = newBonusValue / stack;//newBonusValue: 200, stack: 100, newBonusValuePerItem = 2
+			item.value += newBonusValuePerItem - lastBonusValuePerItem;//newBonusValuePerItem: 2, lastBonusValuePerItem, 20, item.value += -18
+			lastBonusValue = newBonusValue;//newBonusValue: 200, lastBonusValue = 200
+		}
+
+		private bool CheckStackZero(Item item) {
+			if (stack <= 0) {
+				item.value = ContentSamples.ItemsByType[item.type].value;
+				lastBonusValue = 0;
+				return true;
+			}
+
+			return false;
+		}
+
+		public override void ModifyWeaponDamage(Item item, Player player, ref StatModifier damage) {
+			//Gain 1% multiplicative damage for every level on the weapon.
+			damage *= 1f + (float)level / 100f;
+		}
+
+		public override void ModifyTooltips(Item item, List<TooltipLine> tooltips) {
+			if (experience > 0) {
+				tooltips.Add(new TooltipLine(Mod, "level", $"Level: {level}") { OverrideColor = Color.LightGreen });
+				string levelString = $" ({(level + 1) * experiencePerLevel - experience} to next level)";
+				tooltips.Add(new TooltipLine(Mod, "experience", $"Experience: {experience}{levelString}") { OverrideColor = Color.White });
+			}
+		}
+
+		public override void OnCreate(Item item, ItemCreationContext context) {
+			if (context is RecipeCreationContext rContext) {
+				foreach (Item ingredient in rContext.ConsumedItems) {
+					if (ingredient.TryGetGlobalItem(out WeaponWithGrowingDamage ingredientGlobal)) {
+						//Transfer all experience from consumed items to the crafted item.
+						GainExperience(item, ingredientGlobal.experience);
+					}
+				}
+			}
+		}
+
+		public override void OnStack(Item item1, Item item2, int numberToBeTransfered) {
+			if (!item1.TryGetGlobalItem(out WeaponWithGrowingDamage weapon1) || !item2.TryGetGlobalItem(out WeaponWithGrowingDamage weapon2))
+				return;
+
+			if (item1.stack == 0)
+				weapon1.experience = 0;
+			
+			//Transfer experience and value to item1.
+			weapon1.experience += weapon2.experience;//Works
+			//experience += weapon2.experience;//Doesn't work
```

Comment:
remove this comment and the `//Works` one above

Reply:
removed